### ISSUE TITLE
'NoneType' object has no attribute 'rfind' in posixpath.py (when documentation in doc, not docs)

### ISSUE THREAD

**Issue by pllim** [astropy/package-template#157]
I see this error after I applied package template to Ginga (ejeschke/ginga#313). I don't understand where this is coming from. Has anyone else encountered this? If so, what is the solution?

https://travis-ci.org/ejeschke/ginga/builds/119364140

```
Traceback (most recent call last):

  File "setup.py", line 120, in <module>

    **package_info

  File "/home/travis/miniconda/envs/test/lib/python2.7/distutils/core.py", line 151, in setup

    dist.run_commands()

  File "/home/travis/miniconda/envs/test/lib/python2.7/distutils/dist.py", line 953, in run_commands

    self.run_command(cmd)

  File "/home/travis/miniconda/envs/test/lib/python2.7/distutils/dist.py", line 972, in run_command

    cmd_obj.run()

  File "/home/travis/miniconda/envs/test/lib/python2.7/site-packages/astropy/tests/command.py", line 158, in run

    self._build_temp_install()

  File "/home/travis/miniconda/envs/test/lib/python2.7/site-packages/astropy/tests/command.py", line 213, in _build_temp_install

    os.path.basename(self.docs_path))

  File "/home/travis/miniconda/envs/test/lib/python2.7/posixpath.py", line 114, in basename

    i = p.rfind('/') + 1

AttributeError: 'NoneType' object has no attribute 'rfind'

The command "python setup.py $SETUP_CMD" exited with 1.

0.00s$ if [[ $SETUP_CMD == build_sphinx* ]]; then linkchecker doc/_build/html/index.html --ignore-url py-modindex.html; fi
```

**Comment by pllim** [comment]
So it turns out the package template freaks out when documentations are stored in `doc` instead of `docs`. Now the question is, how do I customize this template so that `python setup.py test` automatically sets `docs_path` to `'doc'` and not `'docs'`? It does not seem to know that I have set the following metadata in `setup.cfg`:

```
[build_sphinx]
source-dir = doc
build-dir = doc/_build
all_files = 1
```

**Comment by pllim** [comment]
Fixed via astropy/astropy#4748
